lib/util/glob.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * @fileoverview An inherited `glob.GlobSync` to support .gitignore patterns.
+ * @author Kael Zhang
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var Sync = require("glob").GlobSync,
+    path = require("path"),
+    util = require("util");
+
+//------------------------------------------------------------------------------
+// Private
+//------------------------------------------------------------------------------
+
+var IGNORE = typeof Symbol === "function" ? Symbol("ignore") : "_shouldIgnore";
+
+/**
+ * Subclass of `glob.GlobSync`
+ * @param {string}     pattern      Pattern to be matched.
+ * @param {Object}     options      `options` for `glob`
+ * @param {function()} shouldIgnore Method to check whether a directory should be ignored.
+ * @constructor
+ */
+function GlobSync(pattern, options, shouldIgnore) {
```

Comment:
We cant. `eslint` still targets node 4 which does not support es6 classes. 

Although I usually write packages for node 6 in es8, such as typed variables with the help of babel.  😃